### PR TITLE
Changed notation from surface_waves to stokes_drift

### LINKED ISSUE #1331: Change `IncompressibleModel.surface_waves` to `IncompressibleModel.stokes_drift`?
Currently `IncompressibleModel` has a property called `surface_waves`:

https://github.com/CliMA/Oceananigans.jl/blob/908fc5347c4f3238474d63d10c9314719a79d079/src/Models/IncompressibleModels/incompressible_model.jl#L27

Considering that we may add `free_surface` soon (which involves an explicit free-surface field), do we want to change this model property to `stokes_drift`?

### PR DESCRIPTION
**Pull request by tomchor**
Changed every instance of `surface_waves` to `stokes_drift`.

Closes #1331

### CODE REVIEW

**Comment by ali-ramadhan** [comment]
Ah I think Buildkite tests did not run on this PR since we disable Buildkite on PRs from forks (to avoid random PRs executing potentially malicious code on our machines).

So we usually open and merge branches from the repo itself. You should be able to do this since you're a repo collaborator.

**Comment by tomchor** [comment]
Any way of forcing them to run now so that I don't have to re-do the branch?

**Comment by ali-ramadhan** [comment]
Tried starting a new build manually from your fork but that didn't work (https://buildkite.com/clima/oceananigans/builds/1068) so I enabled Buildkite builds from forks and pushed an empty commit to this PR.

Looks like tests are running now!

File: test/test_surface_waves.jl [hunk 1]
```diff
@@ -3,7 +3,7 @@ function instantiate_surface_waves()
     ∂t_vˢ(z, t) = exp(z/20) * cos(t)
     ∂z_uˢ(z, t) = exp(z/20) * cos(t)
     ∂z_vˢ(z, t) = exp(z/20) * cos(t)
-    surface_waves = SurfaceWaves.UniformStokesDrift(∂t_uˢ=∂t_uˢ, ∂t_vˢ=∂t_vˢ,
+    stokes_drift = SurfaceWaves.UniformStokesDrift(∂t_uˢ=∂t_uˢ, ∂t_vˢ=∂t_vˢ,
                                                     ∂z_uˢ=∂z_uˢ, ∂z_vˢ=∂z_vˢ)
```
Comment (glwagner) [inline test/test_surface_waves.jl:7]:
```suggestion
                                                   ∂z_uˢ=∂z_uˢ, ∂z_vˢ=∂z_vˢ)
```

fixes the alignment issue.

**Review by glwagner — APPROVED** [review]
Thanks @tomchor 🙇‍♂️

**Comment by glwagner** [comment]
Ah, wait. I think we should also change the name of the module from `SurfaceWaves` to `StokesDrift`. @tomchor can you do that?

**Comment by glwagner** [comment]
As a side note, I wish we could use the word "pseudomomentum" rather than Stokes drift. But alas, despite being more correct and far less prone to mislead people about the underlying physics, I'm worried its too obscure :-/

File: test/test_stokes_drift.jl [hunk 2]
```diff
@@ -1,18 +1,18 @@
-function instantiate_surface_waves()
+function instantiate_stokes_drift()
     ∂t_uˢ(z, t) = exp(z/20) * cos(t)
     ∂t_vˢ(z, t) = exp(z/20) * cos(t)
     ∂z_uˢ(z, t) = exp(z/20) * cos(t)
     ∂z_vˢ(z, t) = exp(z/20) * cos(t)
-    stokes_drift = SurfaceWaves.UniformStokesDrift(∂t_uˢ=∂t_uˢ, ∂t_vˢ=∂t_vˢ,
+    stokes_drift = StokesDrift.UniformStokesDrift(∂t_uˢ=∂t_uˢ, ∂t_vˢ=∂t_vˢ,
                                                     ∂z_uˢ=∂z_uˢ, ∂z_vˢ=∂z_vˢ)
```
Comment (glwagner) [inline test/test_stokes_drift.jl:7]:
```suggestion
                                                  ∂z_uˢ=∂z_uˢ, ∂z_vˢ=∂z_vˢ)
```

fixes indenting.

**Pull request merged**